### PR TITLE
fix/preview-controls

### PR DESCRIPTION
**Pull request by ashwins93**
## Details

Do not show controls for mute/unmute if the role isn't allowed to publish

### Choose one of these(put a 'x' in the bracket):
- [x] The change doesn't require a change to the documentation.
- [ ] The documentation is updated accordingly.

### CODE REVIEW

File: src/components/Preview/Preview.tsx [hunk 1]
```diff
@@ -117,11 +118,16 @@ export const Preview = ({
 
   const audioEnabled = useHMSStore(selectIsLocalAudioEnabled);
   const videoEnabled = useHMSStore(selectIsLocalVideoDisplayEnabled);
+  const localPeerRole = useHMSStore(selectLocalPeerRole);
+  const audioAllowed =
```
Comment (raviteja83) [inline src/components/Preview/Preview.tsx:122]: selectIsAllowedToPublish use this selector instead to get audio/video allowed permission

Comment (ashwins93) [reply src/components/Preview/Preview.tsx]: Done

File: src/components/Preview/Preview.tsx [hunk 2]
```diff
@@ -131,7 +133,13 @@ export const Preview = ({
   }, [error.title]);
 
   useEffect(() => {
-    hmsActions.preview(config);
+    try {
+      hmsActions.preview(config).catch((e: any) => {
```
Comment (raviteja83) [inline src/components/Preview/Preview.tsx:137]: .catch should be sufficient I think. try catch is not necessary. Also can we make the log more descriptive

**Review by raviteja83 — DISMISSED** [review]

File: src/components/Preview/Preview.tsx (hunk 2 above)
Comment (ashwins93) [reply src/components/Preview/Preview.tsx]: Accidental commit - reverted now. Preview needs a better error handling. I'll commit this as a part of a separate PR. Please ignore this for now.

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**